![ONT_logo](/ONT_logo.png)
-----------------------------

Pipeline for de novo clustering of long transcriptomic reads
=============================================================



Getting Started
===============

## Input

- The input reads must be in fastq format. 

## Output

The main output files generated by the pipeline are the following:


## Depedencies

- [miniconda](https://conda.io/miniconda.html)
- [snakemake](http://snakemake.readthedocs.io/en/latest/) - easily installed via conda
- The rest of the dependencies are installed via conda.

## Installation

Clone the pipeline and the pipeline toolset by issuing:

```bash
git clone --recursive https://github.com/nanoporetech/pipeline-nanopore-denovo-isoforms.git
```

## Usage

Edit `config.yml` to set the input fastq and parameters, then issue:

```bash
snakemake -j <num_cores> all
```

Results
=======

## Performance on PCS109 SIRV data

![bench_SIRV](/evaluation/results/isonclust2_SIRV.png)

## Performance on PCS109 *Drosophila melanogaster* data

![bench_Dmel](/evaluation/results/isonclust2_dmel.png)

## Licence and Copyright

(c) 2020 Oxford Nanopore Technologies Ltd.

This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this
file, You can obtain one at http://mozilla.org/MPL/2.0/.

## FAQs and tips

## References and Supporting Information

See the post announcing transcriptomics tools in the Nanopore Community [here](https://community.nanoporetech.com/posts/new-transcriptomics-analys).

### Research Release

Research releases are provided as technology demonstrators to provide early access to features or stimulate Community development of tools. Support for this software will be minimal and is only provided directly by the developers. Feature requests, improvements, and discussions are welcome and can be implemented by forking and pull requests. However much as we would like to rectify every issue and piece of feedback users may have, the developers may have limited resource for support of this software. Research releases may be unstable and subject to rapid iteration by Oxford Nanopore Technologies.
